// Copyright 2015 The gRPC Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

option java_multiple_files = true;
option java_package = "com.xuebingli.proto";
option java_outer_classname = "DimProto";

package neural;

service Reduce {
    rpc ReduceDimention (ReduceRequest) returns (ReduceReply) {
    }
}

message ReduceRequest {
    Dataset dataset = 1;
    Algorithm algorithm = 2;
    int32 number = 3;
    int32 dimention = 4;
}

message ReduceReply {
    ReduceRequest request = 1;
    repeated Point2D points2 = 2;
    repeated Point3D points3 = 3;
}

message Point2D {
    float x = 1;
    float y = 2;
}

message Point3D {
    float x = 1;
    float y = 2;
    float z = 3;
}

enum Algorithm {
    PCA = 0;
    TSNE = 1;
}

enum Dataset {
    MNIST = 0;
}
